### PR TITLE
worker: GeoTIFF Optimizer

### PR DESCRIPTION
**Pull request by JakobMiksch**
- worker that makes a GeoTIFF cloud-optimized aka converts it to a COG
- add overviews to the COG

### CODE REVIEW

File: src/packagesToBuild.json [hunk 1]
```diff
@@ -1 +1 @@
-["download-file","gunzip-file","geoserver-publish-sld","geoserver-publish-layer-from-db","geonetwork-publish-metadata","send-email","dispatcher","zip-handler","upload-file", "geoserver-publish-geotiff", "geotiff-validator", "send-mattermost-message", "geoserver-publish-imagemosaic", "geoserver-create-imagemosaic-datastore", "create-file"]
+["download-file","gunzip-file","geoserver-publish-sld","geoserver-publish-layer-from-db","geonetwork-publish-metadata","send-email","dispatcher","zip-handler","upload-file", "geoserver-publish-geotiff", "geotiff-validator", "send-mattermost-message", "geoserver-publish-imagemosaic", "geoserver-create-imagemosaic-datastore", "create-file", "geotiff-optimize"]
```
Comment (JakobMiksch) [inline src/packagesToBuild.json:1]:
```suggestion
["download-file","gunzip-file","geoserver-publish-sld","geoserver-publish-layer-from-db","geonetwork-publish-metadata","send-email","dispatcher","zip-handler","upload-file", "geoserver-publish-geotiff", "geotiff-validator", "send-mattermost-message", "geoserver-publish-imagemosaic", "geoserver-create-imagemosaic-datastore", "create-file", "geotiff-optimizer"]
```

File: src/geotiff-optimizer/index.spec.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+process.env.RABBITHOST = '46.4.8.29';
```
Comment (hblitza) [inline src/geotiff-optimizer/index.spec.js:1]: unsure here. can it be set dynamically?

File: src/geotiff-optimizer/optimize-geotiff.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+            resolve(stdout ? stdout : stderr);
+        });
+    });
+}
+
+/**
+ * Optimize a GeoTIFF for the cloud by converting it to a COG.
+ *
+ * Also computes raster overviews.
+ * Relies on GDAL >= 3.4
```
Comment (hblitza) [inline src/geotiff-optimizer/optimize-geotiff.js:28]: the github action, test.yaml we install gdal via `apt install gdal-bin`.  Will it be > 3.4?

**Comment by hblitza** [comment]
Nice work!

File: src/geotiff-optimizer/index.spec.js (hunk 2 above)
Comment (JakobMiksch) [reply src/geotiff-optimizer/index.spec.js:1]: good point. so far we use this approach everywhere https://github.com/klips-project/rabbitmq-worker/blob/main/src/download-file/index.spec.js#L1

File: src/geotiff-optimizer/optimize-geotiff.js (hunk 3 above)
Comment (JakobMiksch) [reply src/geotiff-optimizer/optimize-geotiff.js:28]: good point. I looked it up: https://packages.ubuntu.com/search?suite=jammy&searchon=names&keywords=gdal
Indeed Ubuntu `22.04` will install GDAL `3.4.1`

**Review by JakobMiksch — COMMENTED** [review]
@hblitza thx for the review. See my response inline

File: src/geotiff-optimizer/index.spec.js (hunk 2 above)
Comment (hblitza) [reply src/geotiff-optimizer/index.spec.js:1]: ok for now. let's change it to a github secret variable or so in follow up pr

**Review by hblitza — APPROVED** [review]

**Pull request merged**